nifi-nar-bundles/nifi-standard-bundle/nifi-standard-parameter-providers/src/main/java/org/apache/nifi/parameter/DatabaseParameterProvider.java

```diff
@@ -0,0 +1,254 @@
+
+        final DBCPService dbcpService = context.getProperty(DBCP_SERVICE).asControllerService(DBCPService.class);
+        final String whereClause = context.getProperty(SQL_WHERE_CLAUSE).getValue();
+        final String parameterNameColumn = context.getProperty(PARAMETER_NAME_COLUMN).getValue();
+        final String parameterValueColumn = context.getProperty(PARAMETER_VALUE_COLUMN).getValue();
+        final String parameterGroupNameColumn = context.getProperty(PARAMETER_GROUP_NAME_COLUMN).getValue();
+
+        final List<String> tableNames = groupByColumn
+                ? Collections.singletonList(context.getProperty(TABLE_NAME).getValue())
+                : Arrays.stream(context.getProperty(TABLE_NAMES).getValue().split(",")).map(String::trim).collect(Collectors.toList());
+
+        final Map<String, List<Parameter>> parameterMap = new HashMap<>();
+        for (final String tableName : tableNames) {
+            try (final Connection con = dbcpService.getConnection(Collections.emptyMap()); final Statement st = con.createStatement()) {
+                final List<String> columns = new ArrayList<>();
+                columns.add(parameterNameColumn);
+                columns.add(parameterValueColumn);
+                if (groupByColumn) {
+                    columns.add(parameterGroupNameColumn);
+                }
+                final String query = getQuery(context, tableName, columns, whereClause);
+
+                getLogger().info("Fetching parameters with query: " + query);
+                try (final ResultSet rs = st.executeQuery(query)) {
+                    while (rs.next()) {
+                        final String parameterName = Objects.requireNonNull(rs.getString(parameterNameColumn), "Parameter Name may not be null");
+                        final String parameterValue = Objects.requireNonNull(rs.getString(parameterValueColumn), "Parameter Value may not be null");
+                        final String parameterGroupName;
+                        if (groupByColumn) {
+                            parameterGroupName = parameterGroupNameColumn == null ? null : rs.getString(parameterGroupNameColumn);
+                        } else {
+                            parameterGroupName = tableName;
+                        }
+
+                        final ParameterDescriptor parameterDescriptor = new ParameterDescriptor.Builder()
+                                .name(parameterName)
+                                .build();
+                        final Parameter parameter = new Parameter(parameterDescriptor, parameterValue);
+                        parameterMap.computeIfAbsent(parameterGroupName, key -> new ArrayList<>()).add(parameter);
+                    }
+                }
+            } catch (final SQLException e) {
+                throw new RuntimeException("Encountered a database error when fetching parameters: " + e.getMessage(), e);
```

Comment:
Is there a better way to surface this error? I get the "Unexpected error occurred" page and have to look in nifi-user.log to figure out what went wrong. Can we put up a bulletin or something in the UI for the user?